eel/src/event_handler.rs

```diff
@@ -60,7 +59,28 @@ impl EventHandler for LipaEventHandler {
 
                 let payment_store = self.payment_store.lock().unwrap();
 
-                if let Ok(payment) = payment_store.get_payment(&payment_hash.0) {
+                if let Ok(payment) = payment_store.get_payment(&payment_hash.0.to_hex()) {
+                    let payment = match payment.should_be_set_expired() {
+                        false => payment,
+                        true => {
+                            match payment_store
+                                .new_payment_state(&payment.hash, PaymentState::InvoiceExpired)
+                            {
+                                Ok(()) => match payment_store.get_payment(&payment.hash) {
+                                    Ok(p) => p,
+                                    Err(e) => {
+                                        error!("Failed find payment with hash {} when previous attempts have succeeded: {}", payment.hash, e);
```

Comment:
Here you handle the bad case, but in the _previous_ attempt that you reference to, the bad case is actually being ignored. ( `if let Ok(payment)` )

Reply:
This was removed while addressing another comment. But that's right, at the moment, we claim htlcs even if there's no associated `Payment` in the db. Should we change that? Most likely, it's an htlc from a spontaneous payment. I propose that if we want to change that, we do so in another PR.

Reply:
Oh okay. I think it's fine.